### ISSUE TITLE
Make fips_to_abbr behave in a more predictable way

### ISSUE THREAD

**Issue by capnrefsmmat** [cmu-delphi/covidcast#421]
Example:

```
> fips_to_abbr("10")
01000 
 "AL" 
Warning message:
Some inputs were not uniquely matched; returning only the first match in each case. 

> fips_to_abbr("10", ties_method="all")
[[1]]
01000 10000 11000 21000 31000 41000 51000 
 "AL"  "DE"  "DC"  "KY"  "NE"  "OR"  "VA" 
```

This isn't great. If I have a bunch of two-digit state FIPS codes and want to match them, I need to do this instead:

```
> fips_to_abbr("^10")
10000 
 "DE" 
```

since it's a regular expression. And so if I have a vector of two-digit FIPS codes from some outside source, I need to do

```
> fips_to_abbr(paste0("^", fipsen))
10000 11000 01000 
 "DE"  "DC"  "AL"
```

Instead, `fips_to_abbr` should understand that two-digit FIPS codes are a prefix match and do this for me.

**Comment by statsmaths** [comment]
The function `fips_to_name` has the exact same problem.

**Comment by statsmaths** [comment]
In other words, is there a reason this doesn't provide what is needed (and if not, what's a counter example that does not work as expected):

```
fips_to_abbr <- function(code)
{
  fips <- sprintf("%02d", covidcast::state_census$STATE)
  index <- match(substr(code, 1, 2), fips)
  output <- covidcast::state_census$ABBR[index]
  names(output) <- code
  output
}
```

**Comment by capnrefsmmat** [comment]
> For me, I don't see a use case where I would want anything other than a single exact match of the a two digit fips code or a single exact match of the first 5 digits of a fips code.

I agree; I don't know of any use cases where I would, say, want to partially match on the inner three digits of a FIPS code. I can't think of any use cases we have for any other type of matching. I don't think there's any meaningful hierarchy to FIPS where one would want, say, all FIPS codes starting with 223 -- 22, yes, but 223 is just an arbitrary set of counties.

As for your suggested code, maybe the only thing I'd change is how the output vector is named:

```
> fips_to_abbr("22")
  22 
"LA" 
> fips_to_abbr("2200")
2200 
"LA" 
> fips_to_abbr("22000")
22000 
 "LA" 
```

For consistency we probably want a common naming used for all states, e.g. every state always gets its two-digit name in the vector.

**Comment by statsmaths** [comment]
Okay, perfect. And good idea about the output (I was matching the input, but probably better to just give standardized FIPS).  I will update this for both `fips_to_abbr` and `fips_to_name` and open a new PR.

**Comment by statsmaths** [comment]
Spoke too soon. `fips_to_abbr` was clear, but I noticed that `fips_to_name` was written to return either the state name or the county name:

```
> fips_to_name("42")
         42000
"Pennsylvania"
> fips_to_name("48429")
            48429
"Stephens County"
```

Given how clearly broken this is (i.e. `fips_to_name("10")` is wrong), it seems doubtful this is used downstream. I also do not see it used in any other **covidcast** functions. Would it make sense to split this into two functions: `fips_to_county_name` and `fips_to_state_name`?

**Comment by capnrefsmmat** [comment]
Here's the only "in the wild" use of `fips_to_name` I can find in 30 seconds: https://github.com/cmu-delphi/www-main/blob/71bbfccbeeb479f2d29bac80dba8aa5c5b597b7f/content/blog/2020-08-26-fb-survey.Rmd#L331-L333

You can see it's trying to go from a county FIPS to a state name, by stripping the first two digits off a FIPS and pasting on 000. It'd probably be better with `fips_to_state_name` (or `state_fips_to_name`?).

I guess there are two options:

* Split into two functions
* If given two digits, match state; if given five, match county (or, if ending in 000, match state)

I expect splitting in two would be less surprising, just because otherwise you have to remember rules about how many digits to use when. But note that the county one still needs to handle FIPS codes ending in 000, because it's natural for someone to do

```
d <- covidcast_signal(..., geo_type = "county")
d$county <- county_fips_to_name(d$geo_value)
```

and if `county_fips_to_name` barfs on FIPS codes ending in 000, it won't handle [megacounties](https://cmu-delphi.github.io/delphi-epidata/api/covidcast_geography.html#small-sample-sizes-and-megacounties).

**Comment by statsmaths** [comment]
I think splitting into two functions makes the most sense. I like your more descriptive naming scheme. And no problem supporting the FIPS ending in 000 (its already in the county data). I will go ahead and make those changes.

**Comment by statsmaths** [comment]
Fixed in #503
